File: apps/nongyu-go-track-server/internal/httpapi/respond.go
package httpapi

import (
	"encoding/json"
	"net/http"
)

type successBody struct {
	OK   bool `json:"ok"`
	Data any  `json:"data"`
}

type failBody struct {
	OK    bool `json:"ok"`
	Error struct {
		Code    string `json:"code"`
		Message string `json:"message"`
	} `json:"error"`
}

func writeOK(w http.ResponseWriter, status int, data any) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(status)
	_ = json.NewEncoder(w).Encode(successBody{OK: true, Data: data})
}

func writeFail(w http.ResponseWriter, status int, code, message string) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(status)
	var body failBody
	body.OK = false
	body.Error.Code = code
	body.Error.Message = message
	_ = json.NewEncoder(w).Encode(body)
}

File: apps/nongyu-node-track-server/src/http/respond.ts
import type { FastifyReply } from "fastify";

export function writeOK(reply: FastifyReply, status: number, data: unknown): void {
  void reply.code(status).type("application/json; charset=utf-8").send({ ok: true, data });
}

export function writeFail(
  reply: FastifyReply,
  status: number,
  code: string,
  message: string,
): void {
  void reply.code(status).type("application/json; charset=utf-8").send({
    ok: false,
    error: { code, message },
  });
}
